test/functional/apm.test.js

```diff
@@ -658,6 +658,46 @@ describe('APM', function () {
     }
   });
 
+  // NODE-1502
+  it('should not allow mutation of internal state from commands returned by event monitoring', {
+    metadata: {
+      /*requires: { mongodb: '>= 3.6.0' } */
+    },
+    test: function () {
+      const self = this;
+      const started = [];
+      const succeeded = [];
+      const client = self.configuration.newClient(
+        { writeConcern: { w: 1 } },
+        { maxPoolSize: 1, monitorCommands: true }
+      );
+      client.on('commandStarted', filterForCommands('insert', started));
+      client.on('commandSucceeded', filterForCommands('insert', succeeded));
+      let documentToInsert = { a: { b: 1 } };
+      return client
+        .connect()
+        .then(client => {
+          const db = client.db(self.configuration.db);
+          return db.collection('apm_test').insertOne(documentToInsert);
+        })
+        .then(r => {
+          expect(r).to.have.property('insertedId').that.is.an('object');
+          expect(started.length).to.equal(1);
+          // Check if contents of returned document are equal to document inserted (by value)
+          expect(documentToInsert).to.deep.equal(started[0].command.documents[0]);
+          // Check if the returned document is a clone of the original. This confirms that the
+          // reference is not the same.
+          expect(documentToInsert !== started[0].command.documents[0]).to.equal(true);
+          expect(documentToInsert.a !== started[0].command.documents[0].a).to.equal(true);
+
+          expect(started[0].commandName).to.equal('insert');
```

Comment:
As far as these assertions are concerned, we want to do these to ensure that the strict equality fails which would mean we're dealing with different object references.
You're right in saying that I could try some mutation and seeing if mutating one affects the other since that would have the same effect as those two tests, but just be more explicit in how it relates to the test name, so I'll get on that.